backend/service/k8s/configmap.go

```diff
@@ -71,11 +72,21 @@ func protoForConfigMap(cluster string, k8sconfigMap *v1.ConfigMap) *k8sapiv1.Con
 		clusterName = cluster
 	}
 
-	return &k8sapiv1.ConfigMap{
+	// required fields
+	ret := &k8sapiv1.ConfigMap{
 		Cluster:     clusterName,
 		Namespace:   k8sconfigMap.Namespace,
 		Name:        k8sconfigMap.Name,
 		Labels:      k8sconfigMap.Labels,
 		Annotations: k8sconfigMap.Annotations,
 	}
+	// optional fields
+	if k8sconfigMap.Data != nil {
+		dataMap := make(map[string]*structpb.Value, len(k8sconfigMap.Data))
+		for k, v := range k8sconfigMap.Data {
+			dataMap[k], _ = structpb.NewValue(v)
```

Comment:
@danielhochman is it okay to absorb this error here?

Reply:
in this case, you can call `NewStringValue` which doesn't return error.

though we should document why we're not unpacking the complex types though and just using the flat string. i'm actually not sure it's the best idea. if we want to be able to read, update, write then the read ideally would return the same format that we end up using for the write. maybe i'm missing something?

Reply:
Reading up a little, realized that we only need to support the map[string]string type for data, its upto the user to pack other values as string. So will go ahead and change the type also also add BinaryData as a field in the proto. Thanks for being patient with this one

Reply:
Updated 👍